git-link.el

```diff
@@ -239,7 +247,8 @@ Return nil,
                       alist))))
 
 (defun git-link--relative-filename ()
-  (let* ((filename (buffer-file-name))
+  (let* ((filename (or (file-remote-p (buffer-file-name) 'localname)
```

Comment:
We support getting links from dired buffers and in this case `buffer-file-name` will be `nil` which will signal an error when passed to `file-or-remote-p`.